mmaction/datasets/pipelines/augmentations.py

```diff
@@ -697,6 +697,141 @@ def __repr__(self):
+
+        self.alpha_std = alpha_std
+        self.color_space_aug = color_space_aug
+
+    @staticmethod
+    def brightnetss(imgs, delta):
+        """Brightness distortion."""
+        if random.uniform(0, 1) > 0.5:
+            delta = np.array(delta).astype(np.float32)
+            imgs = imgs + delta
+        return imgs
+
+    @staticmethod
+    def contrast(imgs, alpha):
+        """Contrast distortion"""
+        if random.uniform(0, 1) > 0.5:
+            alpha = np.array(alpha).astype(np.float32)
+            imgs = imgs * alpha
+        return imgs
+
+    @staticmethod
+    def saturation(imgs, alpha):
+        """Saturation distortion."""
+        if random.uniform(0, 1) > 0.5:
+            gray = imgs * np.array([0.114, 0.587, 0.299]).astype(np.float32)
+            gray = np.sum(gray, 2, keepdims=True)
+            gray *= (1.0 - alpha)
+            imgs = imgs * alpha
+            imgs = imgs + gray
+        return imgs
+
+    @staticmethod
+    def hue(img, alpha):
+        """Hue distortion"""
+        if random.uniform(0, 1) > 0.5:
+            u = np.cos(alpha * np.pi)
+            w = np.sin(alpha * np.pi)
+            bt = np.array([[1.0, 0.0, 0.0], [0.0, u, -w], [0.0, w, u]])
+            tyiq = np.array([[0.299, 0.587, 0.114], [0.596, -0.274, -0.321],
+                             [0.211, -0.523, 0.311]])
+            ityiq = np.array([[1.0, 0.956, 0.621], [1.0, -0.272, -0.647],
+                              [1.0, -1.107, 1.705]])
+            t = np.dot(np.dot(ityiq, bt), tyiq).T
+            t = np.array(t).astype(np.float32)
+            # Here we use np.flip to suit our RGB images cases.
+            t = np.flip(t)
+            img = np.dot(img, t)
+        return img
+
+    def __call__(self, results):
+        imgs = results['imgs']
+        out = []
+        if self.color_space_aug:
+            bright_delta = np.random.uniform(-32, 32)
+            contrast_alpha = np.random.uniform(0.6, 1.4)
+            saturation_alpha = np.random.uniform(0.6, 1.4)
+            hue_alpha = random.uniform(-18, 18)
+            for img in imgs:
+                img = self.brightnetss(img, delta=bright_delta)
+                if random.uniform(0, 1) > 0.5:
+                    img = self.contrast(img, alpha=contrast_alpha)
+                    img = self.saturation(img, alpha=saturation_alpha)
+                    img = self.hue(img, alpha=hue_alpha)
+                else:
+                    img = self.saturation(img, alpha=saturation_alpha)
+                    img = self.hue(img, alpha=hue_alpha)
+                    img = self.contrast(img, alpha=contrast_alpha)
+                out.append(img)
+        else:
+            out = imgs
+        alpha = np.random.normal(0, self.alpha_std, size=(3, ))
+        rgb = np.array(np.dot(self.eig_vec * alpha,
+                              self.eig_val)).astype(np.float32)
+        rgb = np.expand_dims(np.expand_dims(rgb, 0), 0)
+
+        results['imgs'] = [img + rgb for img in out]
```

Comment:
what dtype is the imgs? i guess it is still uint8? will it be transformed into float by accident?

Reply:
and why this rgb noise is shared in the image list?

Reply:
It adds PCA bias to images, like [this](https://mxnet.apache.org/api/python/docs/_modules/mxnet/image/image.html#CreateAugmenter)